tests/conftest.py

```diff
@@ -1,8 +1,30 @@
+import asyncio
+import sys
+
+if sys.version_info >= (3, 7):
+    from contextlib import asynccontextmanager
+else:
+    from async_generator import asynccontextmanager
+
 import pytest
 import trustme
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
 
+from uvicorn import Config, Server
+
+
+@asynccontextmanager
+async def run_server(config: Config, sockets=None):
+    server = Server(config=config)
+    cancel_handle = asyncio.ensure_future(server.serve(sockets=sockets))
+    await asyncio.sleep(0.1)
+    try:
+        yield server
+    finally:
+        await server.shutdown()
+        cancel_handle.cancel()
```

Comment:
I _think_ we already have a `utils.py` module. What about move this there? If not, maybe we can consider it. Importing from conftest feels a bit strange given this is a special pytest module — things only we use are probably better in a separate module.

Reply:
yes you're more than correct, I moved everything (import and run-server) in a new utils.py